### PR TITLE
 Fix off by one error in call to goto-char

### PR DESCRIPTION
**Pull request by nordlow**
When offset is zero `(byte-to-position (string-to-number offset))` errors, otherwise (in most cases) cursor is placed one character to the left of the symbol of interest.

### CODE REVIEW

File: ac-dcd.el [hunk 1]
```diff
@@ -535,7 +535,8 @@ dcd-client outputs candidates that begin with \"this\" when completing struct co
 
 (defun ac-dcd-get-ddoc ()
   "Get document with `dcd-client --doc'."
-  (save-buffer)
+  (interactive)
+  ;; (save-buffer)
```
Comment (atilaneves) [inline ac-dcd.el]: ?

File: ac-dcd.el [hunk 2]
```diff
@@ -545,14 +546,18 @@ dcd-client outputs candidates that begin with \"this\" when completing struct co
 
     (with-current-buffer buf
       (erase-buffer)
-
+	  (apply 'call-process-region (point-min) (point-max)
```
Comment (atilaneves) [inline ac-dcd.el]: ?

File: ac-dcd.el [hunk 3]
```diff
@@ -545,14 +546,18 @@ dcd-client outputs candidates that begin with \"this\" when completing struct co
...
+			 ac-dcd-executable nil buf nil args)
           (apply 'call-process-region (point-min) (point-max)
                          ac-dcd-executable nil buf nil args)
       (when (or
              (string= (buffer-string) "")
              (string= (buffer-string) "\n\n\n")             ;when symbol has no doc
              )
-        (error "No document for the symbol at point!"))
+        (progn
```
Comment (atilaneves) [inline ac-dcd.el]: ?

File: ac-dcd.el [hunk 4]
```diff
@@ -583,18 +588,20 @@ dcd-client outputs candidates that begin with \"this\" when completing struct co
   "Goto the point where `ac-dcd-goto-definition' was last called."
   (interactive)
   (if (ring-empty-p ac-dcd-goto-definition-marker-ring)
-      (error "Marker ring is empty. Can't pop.")
+      (progn (message "Marker ring is empty. Can't pop.")
+             (ding))
     (let ((marker (ring-remove ac-dcd-goto-definition-marker-ring 0)))
       (switch-to-buffer (or (marker-buffer marker)
-                            (error "Buffer has been deleted")))
+                            (progn (message "Buffer has been deleted")
```
Comment (atilaneves) [inline ac-dcd.el]: 2nd time this exact sequence is used. Better to define a function. What's wrong with the `error` function, anyway?

File: ac-dcd.el [hunk 5]
```diff
@@ -605,7 +612,7 @@ dcd-client outputs candidates that begin with \"this\" when completing struct co
         (ac-dcd-goto-def-push-marker)
         (unless (string=  file "stdin") ; the declaration is in the current file
           (find-file file))
-        (goto-char (byte-to-position (string-to-number offset)))))))
+        (goto-char (byte-to-position (1+ (string-to-number offset))))))))
```
Comment (atilaneves) [inline ac-dcd.el]: And this always works?

File: ac-dcd.el [hunk 6]
```diff
@@ -770,5 +777,29 @@ or package.json file."
             (local-set-key "q" 'delete-window)
             (local-set-key (kbd "RET") 'ac-dcd-visit-file-in-line)))))))
 
+;;;###autoload
+(defun d-turn-on-eldoc-mode ()
```
Comment (atilaneves) [inline ac-dcd.el]: Why is the function not namespaced as `ac-dcd-`?

**Comment by nordlow** [comment]
Only the last commit should be included in this pr... Accidentally branched from my master instead of upstream. Shall I redo this PR?

File: ac-dcd.el (hunk 5 above)
Comment (nordlow) [reply ac-dcd.el]: Yes, because dcd uses zero-based offset and `goto-char` expecteds 1-one based offset.

**Comment by atilaneves** [comment]
> Only the last commit should be included in this pr... Accidentally branched from my master instead of upstream. Shall I redo this PR?

Or just rebase and force push.

**Comment by nordlow** [comment]
Closing and redoing a new PR.

**Pull request closed without merging**